src/uct/rocm/base/rocm_base.c

```diff
@@ -223,6 +222,62 @@ ucs_status_t uct_rocm_base_mem_query(uct_md_h md, const void *addr,
     return UCS_OK;
 }
 
+static hsa_status_t uct_rocm_hsa_pool_callback(hsa_amd_memory_pool_t pool, void* data)
+{
+    int allowed;
+    uint32_t flags;
+    hsa_amd_segment_t segment;
+
+    hsa_amd_memory_pool_get_info(pool, HSA_AMD_MEMORY_POOL_INFO_RUNTIME_ALLOC_ALLOWED, &allowed);
+    if (allowed) {
+        hsa_amd_memory_pool_get_info(pool, HSA_AMD_MEMORY_POOL_INFO_SEGMENT, &segment);
+        if (HSA_AMD_SEGMENT_GLOBAL != segment) {
+            return HSA_STATUS_SUCCESS;
+        }
+
+        hsa_amd_memory_pool_get_info(pool, HSA_AMD_MEMORY_POOL_INFO_GLOBAL_FLAGS, &flags);
+        if (flags & HSA_AMD_MEMORY_POOL_GLOBAL_FLAG_COARSE_GRAINED) {
+            *((hsa_amd_memory_pool_t*)data) = pool;
+            return HSA_STATUS_INFO_BREAK;
+        }
+    }
+    return HSA_STATUS_SUCCESS;
+}
+
+ucs_status_t uct_rocm_base_get_link_type(hsa_amd_link_info_type_t *link_type)
+{
+    hsa_amd_memory_pool_link_info_t link_info;
+    hsa_agent_t agent1, agent2;
+    hsa_amd_memory_pool_t pool;
+    hsa_status_t status;
+
+    *link_type = HSA_AMD_LINK_INFO_TYPE_PCIE;
+
+    if (uct_rocm_base_agents.num_gpu < 2) {
+        return UCS_OK;
+    }
+
+    agent1 = uct_rocm_base_agents.gpu_agents[0];
+    agent2 = uct_rocm_base_agents.gpu_agents[1];
+
+    status = hsa_amd_agent_iterate_memory_pools(agent2,
+                            uct_rocm_hsa_pool_callback, (void*)&pool);
+    if (status != HSA_STATUS_SUCCESS && status != HSA_STATUS_INFO_BREAK) {
```

Comment:
codestyle: please use `()` for each statement:
`if ((status != HSA_STATUS_SUCCESS) && (status != HSA_STATUS_INFO_BREAK))`

Reply:
Fixed